Modules/SubModule_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -244,15 +244,19 @@ function xASL_wrp_ResampleASL(x)
     elseif x.modules.asl.bTimeEncoded
         % Decoding of TimeEncoded data (Nifti is saved inside the function)
         ASL_im = xASL_quant_HadamardDecoding(PathASL{iSpace}, x.Q);
+        HadamardBlock = size(ASL_im,4) / (x.Q.TimeEncodedMatrixSize-1);

```

Comment:
I am quite lost here what this means and how this is later used. Can you please improve comments in lines 247-257 and always very clearly say what should each variable contain?

HadamardBlock contains the number of repetitions of the block or possibly also repetitions*nTE (or rather only nTE, since repetitions are averaged in Decoding), so this will be always >=1. Then on line 249 - this condition will never be satisfied. For once, number of volume is always <= number of unique PLDs, and always < if you add HadamardBlock, which should be always >=1. 

I would need more explanation to understand this. Possibly fix the error, and more specific comments at each line to explain.